### PR TITLE
Component retryable errors for install/upgrade

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Creates a custom error type to be returned from `spi.Component` to indicate to the controller that it should requeue, and is a non-fatal event. 

Also, adds support to `bom.KeyValue` and `HelmComponent` to allow a component to specify a custom helm file override.

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/spi/errors.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/spi/errors.go:1]: I think this is useful for all the controller in both operators.  Maybe put it in top level pkg.

File: platform-operator/controllers/verrazzano/component/spi/errors.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package spi
+
+import "fmt"
+
+type RetryableError struct {
+	Component string
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/spi/errors.go:8]: I thought this  struct would just have the original error plus a ctrl.Result{}.

Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/spi/errors.go:8]: OK, works for me.  I wasn't sure if folks wanted to record their errors locally, and if so if it made sense to propagate those up.

Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/spi/errors.go:12]: Pushed a second pass, where an optional error and/or result can be returned.

**Comment by mbenmclean** [comment]
So does this change pass in the file overrides with -f instead of --set-file? That's what it looks like to me. Do you know if these do the same thing?

**Review by mbenmclean — DISMISSED** [review]

**Review by mbenmclean — APPROVED** [review]

**Pull request merged**